File: src/main/java/ru/hollowhorizon/hollowengine/common/entities/ModEntities.java
package ru.hollowhorizon.hollowengine.common.entities;

import net.minecraft.entity.EntityClassification;
import net.minecraft.entity.EntityType;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import ru.hollowhorizon.hc.api.registy.HollowRegister;
import ru.hollowhorizon.hollowengine.HollowEngine;
import ru.hollowhorizon.hollowengine.client.render.enitites.SindyRenderer;

public class ModEntities {
    public static final DeferredRegister<EntityType<?>> ENTITIES = DeferredRegister.create(ForgeRegistries.ENTITIES, HollowEngine.MODID);
    public static final RegistryObject<EntityType<NPCEntity>> NPC_ENTITY = ENTITIES.register("npc_entity", () -> EntityType.Builder.of(
            (EntityType.IFactory<NPCEntity>) NPCEntity::new,
            EntityClassification.CREATURE).sized(0.6F, 1.8F).build("hollowengine:npc_entity"));
    public static final RegistryObject<EntityType<ThomasNPC>> THOMAS = ENTITIES.register("thomas", () -> EntityType.Builder.of(
            ThomasNPC::new,
            EntityClassification.CREATURE).sized(0.6F, 1.8F).build("hollowengine:thomas"));

    @HollowRegister(renderer = "ru.hollowhorizon.hollowengine.client.render.enitites.SindyRenderer")
    public static final EntityType<SindyNPC> SINDY = EntityType.Builder.of(SindyNPC::new, EntityClassification.CREATURE).sized(0.6F, 1.8F).build("hollowstory:sindy");

    public static void register() {

    }
}

File: common/src/main/java/ru/hollowhorizon/hollowengine/common/registry/ModEntities.kt
package ru.hollowhorizon.hollowengine.common.registry

import net.minecraft.resources.ResourceLocation
import net.minecraft.world.entity.EntityType
import net.minecraft.world.entity.Mob
import net.minecraft.world.entity.MobCategory
import ru.hollowhorizon.hc.common.events.SubscribeEvent
import ru.hollowhorizon.hc.common.events.registry.RegisterEntityAttributesEvent
import ru.hollowhorizon.hc.common.registry.HollowRegistry
import ru.hollowhorizon.hc.common.registry.RegistryObject
import ru.hollowhorizon.hollowengine.HollowEngine
import ru.hollowhorizon.hollowengine.common.entities.NPCEntity

object ModEntities : HollowRegistry() {
    val NPC_ENTITY: RegistryObject<EntityType<NPCEntity>> by register(
        ResourceLocation.fromNamespaceAndPath(HollowEngine.MODID, "npc_entity")
    ) {
        EntityType.Builder.of(::NPCEntity, MobCategory.CREATURE).sized(0.6f, 1.8f).build("npc_entity")
    }
}

@SubscribeEvent
fun onRegisterAttributes(event: RegisterEntityAttributesEvent) {
    event.register(ModEntities.NPC_ENTITY.get(), Mob.createMobAttributes().build())
}